external_codes/catch/complex_approx.hpp

```diff
@@ -9,19 +9,19 @@ namespace Catch {
 class ComplexApprox
 {
 public:
-    ComplexApprox(const std::complex<double> &value) : m_value(value), m_compare_real_only(false) {
+    ComplexApprox(const std::complex<double> &value) : m_value(value) {
       init_epsilon();
     }
 
-    ComplexApprox(const std::complex<float> &value) : m_value(value), m_compare_real_only(false) {
+    ComplexApprox(const std::complex<float> &value) : m_value(value) {
```

Comment:
Is it possible to reduce the four constructors to two template constructors, one for complex and one for real?

Reply:
Maybe, but that feels like a change for a different PR.

Reply:
Could you try it locally? If many other changes are needed, then put it on a separate PR.

Reply:
I tried it locally, it works.   However, part of the reason for so many constructors is a bug in the Intel compiler.   I'm not really interested in doing the testing.   You can make the change if you feel it is important enough, and want to do the testing.

Reply:
If I remember correctly, Intel compiler failed if there is only `ComplexApprox(std::complex<double> value)`. If we use `ComplexApprox(std::complex<T> value)`, it will probably work and we can save a few lines of code. However, the real issue of the ComplexApprox is that it is part of QMCPACK not Catch. If we have unit tests rely on it, itself must be tested. We don't have any unit tests for it so far.

Reply:
For sure, adding tests should be a separate PR.

Reply:
OK. This seems not critical. Let us tackle it when it becomes an issue or we need to work on this part of code. We may use it beyond just `float` and `double` .